### DISCUSSION TITLE
Issue with Tab Management in SeleniumBase

### DISCUSSION THREAD

**Discussion by benfrankie** [seleniumbase/SeleniumBase#2329] [Q&A] [answered]
Hi,

I've encountered a behavior in SeleniumBase that doesn't occur with regular Selenium. When using SeleniumBase to open a new tab and perform actions on the same domain, the script unexpectedly switches back to the original tab. This issue doesn't arise when using plain Selenium.

Here's a brief overview:

- Open a website (e.g., reddit.com) in the main tab.
- Open a new tab and navigate to the same website.
- Attempt to perform actions on the new tab.
- In SeleniumBase, the driver switches back to the main tab immediately, while in regular Selenium, it performs as expected.

Could this be related to SeleniumBase's internal handling of tabs, especially when the same domain is accessed in multiple tabs? What's the solution here while using UC mode?

Here's a snippet of the test code that illustrates the issue:
```python
driver.get("https://www.reddit.com/")
main_window_handle = driver.current_window_handle
# Open new tab and switch to it
driver.execute_script("window.open('');")
new_tab_handle = [handle for handle in driver.window_handles if handle != main_window_handle][0]
driver.switch_to.window(new_tab_handle)

# Attempt to perform actions in the new tab
driver.get("https://www.reddit.com/")
# Intended actions here
smooth_random_scroll(driver)
```

This works as expected for some websites (like "https://yahoo.com/"), but not for others (like "https://www.reddit.com/").

Thanks!

**Comment by mdmintz** [comment, marked as answer]
Even regular `undetected-chromedriver` didn't let you use the standard tab-switching methods as is:
* https://github.com/ultrafunkamsterdam/undetected-chromedriver/issues/368#issuecomment-975993293
* https://github.com/ultrafunkamsterdam/undetected-chromedriver/issues/380#issuecomment-985986285

Tab-switching works differently in UC Mode. (The standard SeleniumBase formats have additional methods for that.)

Here's an example of that:

```python
from seleniumbase import SB

with SB(uc=True) as sb:
    sb.driver.uc_open_with_tab("data:text/html,<h1>Page A</h1>")
    sb.assert_text("Page A")
    sb.open_new_window()
    sb.driver.uc_open_with_tab("data:text/html,<h1>Page B</h1>")
    sb.assert_text("Page B")
    sb.switch_to_window(0)
    sb.assert_text("Page A")
    sb.assert_text_not_visible("Page B")
    sb.switch_to_window(1)
    sb.assert_text("Page B")
    sb.assert_text_not_visible("Page A")
```

If you need more than one tab open, first call `sb.open_new_window()`, then call `sb.driver.uc_open_with_tab(url)`. This will maintain window ordering and numbering.

Also note that having more than one tab open at a time will likely lead to Selenium detection, so you're probably better off having only one tab/window open at a time for tests.

**Reply by benfrankie** [reply]
Hi, I just noticed that a new tab now opens when using SeleniumBase in UC mode. I'm wondering if there's been an update to enable this, and if having more than one tab open at a time will still likely lead to Selenium detection? @mdmintz 
Thanks!

**Reply by mdmintz** [reply]
@benfrankie When opening a new URL in UC Mode, a new tab should open and the old one should close. That's how to be sure that there are no selenium-specific variables set in the browser. It's always been like that. Multiple tabs would likely lead to detection.
